src/hash.hpp

```diff
@@ -13,115 +15,138 @@
 #include "common.h"
 #include "cyclic.hpp"
 
-#include <vector>
-
 namespace Simhash {
-    /**
-     * Compute a similarity hash for the provided string using a rolling hash
-     * function and a provided hash window
-     *
-     * @param tokens - An NULL-terminated array of character pointers to the
-     *                 nul-terminated tokens comprising the text.
-     *
-     * @return hash representative of the content of the text */
-    template <typename Hash=jenkins>
-    class Simhash {
+    class HasherInterface {
+    public:
+        virtual void update(hash_t) = 0;
```

Comment:
Dan and I discussed the template vs non-template issue. We agreed to leave it as is for now. We'll revisit later if it becomes a problem.